api/src/main/java/org/apache/iceberg/RewriteManifests.java

```diff
@@ -49,9 +51,35 @@
    * that do not match the predicate are kept as-is. If this is not called and no predicate is set, then
    * all manifests will be rewritten.
    *
+   * Note that either clusterBy/rewriteIf should be used or deleteManifest/addManifest, not both.
```

Comment:
Why can't we keep the two separate?

We could exclude any deleted manifests from the clusterBy rewrite, and append all new manifests each time.

Reply:
@rdblue not sure I got. Are you suggesting to put them in separate classes or actually make them work with one another?

Reply:
Sorry, I wasn't very clear. I was thinking about making them work together by separating the manifests into two groups: explicit rewrites (using `deleteManifest` and `addManifest`) and clustering rewrites (using `clusterBy` and `rewriteIf`). It would work like this:

1. Don't rewrite manifests that are deleted or added with clustering
2. If there is no `clusterBy` callback, then only apply the adds and deletes

That way, the two work at the same time. If this seems too confusing for users, then we can separate them into different operations, but I think it would be okay.

Reply:
@rdblue, I have been thinking more about this and maybe @bryanck's idea is the easiest to perceive for users. If we develop a common class and then provide `RewriteManifests` and `ReplaceManifests` operations, we won't put too much in `RewriteManifests` and the correct usage will be more obvious. What do you think? The only drawback is one more action in the `Table` API but it seems worth it in this case.

Reply:
I made both of them work at the same time.